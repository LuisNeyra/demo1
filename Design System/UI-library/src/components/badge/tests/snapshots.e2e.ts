import { snapIt } from '../../../test';

const component = 'badge';

describe(component, () => {
  const snap = snapIt(component);

  describe('renders', () => {
    snap('renders correctly with no props', '<blaze-badge>default</blaze-badge>');
    snap('appends the color modifier class', '<blaze-badge type="brand">brand</blaze-badge>');
    snap('handles boolean rounded attribute', '<blaze-badge rounded>default</blaze-badge>');
    snap('handles both boolean rounded attribute and type', '<blaze-badge rounded type="brand">brand</blaze-badge>');
    snap('handles boolean ghost and boolean rounded', '<blaze-badge ghost rounded>default</blaze-badge>');
    snap('supports all the modifiers', '<blaze-badge ghost rounded type="brand">brand</blaze-badge>');
  });
});
