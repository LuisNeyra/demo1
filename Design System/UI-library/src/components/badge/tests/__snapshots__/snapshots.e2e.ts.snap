// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`badge renders appends the color modifier class 1`] = `"<blaze-badge type=\\"brand\\" class=\\"hydrated\\"><!----><span class=\\"c-badge c-badge--brand\\">brand</span></blaze-badge>"`;

exports[`badge renders handles boolean ghost and boolean rounded 1`] = `"<blaze-badge ghost=\\"\\" rounded=\\"\\" class=\\"hydrated\\"><!----><span class=\\"c-badge c-badge--rounded c-badge--ghost\\">default</span></blaze-badge>"`;

exports[`badge renders handles boolean rounded attribute 1`] = `"<blaze-badge rounded=\\"\\" class=\\"hydrated\\"><!----><span class=\\"c-badge c-badge--rounded\\">default</span></blaze-badge>"`;

exports[`badge renders handles both boolean rounded attribute and type 1`] = `"<blaze-badge rounded=\\"\\" type=\\"brand\\" class=\\"hydrated\\"><!----><span class=\\"c-badge c-badge--brand c-badge--rounded\\">brand</span></blaze-badge>"`;

exports[`badge renders renders correctly with no props 1`] = `"<blaze-badge class=\\"hydrated\\"><!----><span class=\\"c-badge\\">default</span></blaze-badge>"`;

exports[`badge renders supports all the modifiers 1`] = `"<blaze-badge ghost=\\"\\" rounded=\\"\\" type=\\"brand\\" class=\\"hydrated\\"><!----><span class=\\"c-badge c-badge--brand c-badge--rounded c-badge--ghost\\">brand</span></blaze-badge>"`;
